akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterShardingSettings.scala

```diff
@@ -131,6 +129,72 @@ object ClusterShardingSettings {
   private[akka] def roleOption(role: String): Option[String] =
     if (role == "") None else Option(role)
 
+  final class PassivationStrategySettings(
+      val strategy: String,
+      val idleTimeout: FiniteDuration,
+      val leastRecentlyUsedLimit: Int) {
+
+    def withIdleStrategy(timeout: FiniteDuration): PassivationStrategySettings =
+      copy(strategy = "idle", idleTimeout = timeout)
+
+    def withLeastRecentlyUsedStrategy(limit: Int): PassivationStrategySettings =
+      copy(strategy = "least-recently-used", leastRecentlyUsedLimit = limit)
+
+    private def copy(
+        strategy: String,
+        idleTimeout: FiniteDuration = idleTimeout,
+        leastRecentlyUsedLimit: Int = leastRecentlyUsedLimit): PassivationStrategySettings =
+      new PassivationStrategySettings(strategy, idleTimeout, leastRecentlyUsedLimit)
+  }
+
+  object PassivationStrategySettings {
+    def apply(config: Config): PassivationStrategySettings = {
+      val settings = new PassivationStrategySettings(
+        strategy = config.getString("passivation.strategy").toLowerCase,
+        idleTimeout = config.getDuration("passivation.idle.timeout", MILLISECONDS).millis,
+        leastRecentlyUsedLimit = config.getInt("passivation.least-recently-used.limit"))
+      // default to old setting if it exists (defined in application.conf), overriding the new settings
+      if (config.hasPath("passivate-idle-entity-after")) {
+        val timeout =
+          if (config.getString("passivate-idle-entity-after").toLowerCase == "off") Duration.Zero
+          else config.getDuration("passivate-idle-entity-after", MILLISECONDS).millis
+        settings.withIdleStrategy(timeout)
+      } else {
+        settings
+      }
+    }
+
+    def oldDefault(idleTimeout: FiniteDuration): PassivationStrategySettings =
+      new PassivationStrategySettings("idle", idleTimeout, 0)
+  }
+
+  /**
+   * INTERNAL API
+   */
+  sealed trait PassivationStrategy
+  case object NoPassivationStrategy extends PassivationStrategy
+  case class IdlePassivationStrategy(timeout: FiniteDuration) extends PassivationStrategy
+  case class LeastRecentlyUsedPassivationStrategy(limit: Int) extends PassivationStrategy
+
+  /**
+   * INTERNAL API
+   * Determine the passivation strategy to use from settings.
+   */
+  object PassivationStrategy {
+    def apply(settings: ClusterShardingSettings): PassivationStrategy = {
+      if (settings.rememberEntities) {
+        NoPassivationStrategy
```

Comment:
> Automatic passivation is disabled on remember entities, like before, but do we want to combine remember entities with the limit-based passivation strategies?

I think it is still a good approach to not have any automatic passivation when `rememberEntities` is enabled. The primary purpose of `rememberEntities` is to always have it running and it would be difficult to reason about that if it sometimes starts to automatically passivate those.

Reply:
Sounds good 👍🏼